rust/src/utils/gas.rs

```diff
@@ -7,7 +7,7 @@ use crate::{
 };
 
 #[inline(always)]
-pub fn consume_gas(gas_left: &mut u64, gas: u64) -> Result<(), StatusCode> {
+pub fn consume_gas(gas: u64, gas_left: &mut u64) -> Result<(), StatusCode> {
```

Comment:
Because you decided to change the order of these two parameters, I realized that this is slightly awkward. 
You still need to add `&mut` in the caller side, which makes a little more clear which of the two parameters is the cost and which one is the accumulator. 
I still think that the best would be to have a member function on the gas or context. wdyt?